modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/log/GridLogCommandHandler.java

```diff
@@ -0,0 +1,158 @@
+import java.util.Collection;
+import java.util.List;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.processors.rest.GridRestCommand;
+import org.apache.ignite.internal.processors.rest.GridRestResponse;
+import org.apache.ignite.internal.processors.rest.handlers.GridRestCommandHandlerAdapter;
+import org.apache.ignite.internal.processors.rest.request.GridRestLogRequest;
+import org.apache.ignite.internal.processors.rest.request.GridRestRequest;
+import org.apache.ignite.internal.util.future.GridFinishedFuture;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+import static org.apache.ignite.internal.processors.rest.GridRestCommand.LOG;
+
+/**
+ * Handler for {@link org.apache.ignite.internal.processors.rest.GridRestCommand#LOG} command.
+ */
+public class GridLogCommandHandler extends GridRestCommandHandlerAdapter {
+    /**
+     * Supported commands.
+    */
+    private static final Collection<GridRestCommand> SUPPORTED_COMMANDS = U.sealList(LOG);
+
+    /**
+     * Default log file name *
+     */
+    private static final String DEFAULT_LOG_PATH = "work/log/ignite.log";
+
+    /**
+     * Default log file start line number *
+     */
+    private static final int DEFAULT_FROM = 0;
+
+    /**
+     * Default log file end line number*
+     */
+    private static final int DEFAULT_TO = 1;
+
+    /**
+     * @param ctx Context.
+     */
+    public GridLogCommandHandler(GridKernalContext ctx) {
+        super(ctx);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<GridRestCommand> supportedCommands() {
+        return SUPPORTED_COMMANDS;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IgniteInternalFuture<GridRestResponse> handleAsync(GridRestRequest req) {
+        assert req != null;
+        if (req.command() == LOG) {
+            if (log.isDebugEnabled())
+                log.debug("Handling log REST request: " + req);
+            GridRestLogRequest req0 = (GridRestLogRequest) req;
+            try {
+                validateRange(req0);
+            } catch (IgniteCheckedException e){
+                return new GridFinishedFuture<>(new GridRestResponse(GridRestResponse.STATUS_FAILED, e.getMessage()));
+            }
+            int from = DEFAULT_FROM;
+            if (req0.from() != -1) {
+                from = req0.from();
+            }
+            int to = DEFAULT_TO;
+            if (req0.to() != -1) {
+                to = req0.to();
+            }
+            String path = DEFAULT_LOG_PATH;
+            Path filePath = getPath(req0, path);
+            if (from >= to) {
+                return new GridFinishedFuture<>(new GridRestResponse(GridRestResponse.STATUS_FAILED,
+                        "Log file start from cannot be greater than to"));
+            }
+            if(req0.command() == LOG) {
```

Comment:
No need to check this twice. I've already done this at the beginning of the method.